tests/rwlock.rs

```diff
@@ -49,7 +49,8 @@ fn smoke() {
         let lock = RwLock::new(());
         drop(lock.read().await);
         drop(lock.write().await);
-        drop((lock.read().await, lock.read().await));
```

Comment:
This is behaviorally different than the change. Please undo this. This will wait until both have read access before dropping them.

Reply:
Are you sure about this? clippy said it was a useless call, so it might be a clippy bug if you're right

Reply:
I am absolutely sure.

`(lock.read().await, lock.read().await)` means two immutable `read` locks are acquired and simultaneously alive before drop.

Reply:
Okay, I'll revert this